R/license_note.R

```diff
@@ -0,0 +1,79 @@
+#' Generate LICENSE.note file.
+#'
+#' LICENSE.note generated by this function lists dependent Rust packages infomation.
+#' To use this function, the [cargo-lincense](https://crates.io/crates/cargo-license) command must be installed.
+#' @param force Logical indicating whether to re-generating LICENSE.note if LICENSE.note is already exists.
+#' @inheritParams register_extendr
+#' @return No return value, called for side effects.
+#' @export
+#' @importFrom dplyr %>%
+write_license_note <- function(path = ".", force = TRUE) {
+  if (!requireNamespace("RcppTOML", quietly = TRUE)) {
+    cli::cli_abort(
+      "The {.pkg RcppTOML} package is required to run the {.fun write_license_note} function.",
+      class = "rextendr_error"
+    )
+  }
+
+  manifest_file <- rprojroot::find_package_root_file("src", "rust", "Cargo.toml", path = path)
+  out_file <- rprojroot::find_package_root_file("LICENSE.note", path = path)
+
+  if (!isTRUE(force) && file.exists(out_file)) {
+    cli::cli_abort(
+      c(
+        "LICENSE.note already exists.",
+        "If you want to re-generate LICENSE.note, set `force = TRUE` to {.fun write_license_note}."
+      ),
+      class = "rextendr_error"
+    )
+  }
+
+  package_name <- RcppTOML::parseTOML(manifest_file)$package$name
+
```

Comment:
Could there be a processx call or `cargo-license` before this to check to see if it is available?